File: javascript/tests/median_of_two_sorted_arrays_test.js
import {assertEquals} from "@std/assert";
import {findMedianSortedArrays} from "@/median_of_two_sorted_arrays.js";


Deno.test(function exampleCases() {
    assertEquals(findMedianSortedArrays([1, 3, 8, 9, 15], [7, 11, 18, 19, 21, 25]), 11);
    assertEquals(findMedianSortedArrays([1, 3], [2]), 2);
    assertEquals(findMedianSortedArrays([1, 2], [3, 4]), 2.5);
});

Deno.test(function extraCases() {
    assertEquals(findMedianSortedArrays([1], []), 1);
    assertEquals(findMedianSortedArrays([], [1]), 1);
    assertEquals(findMedianSortedArrays([1], [2]), 1.5);
    assertEquals(findMedianSortedArrays([1, 3, 5], [2, 4]), 3);
    assertEquals(findMedianSortedArrays([1, 3, 4, 5, 6], [2]), 3.5);
});

File: elixir/test/median_of_two_sorted_arrays_test.exs
defmodule MedianOfTwoSortedArraysTest do
  use ExUnit.Case

  import MedianOfTwoSortedArrays

  @moduletag :capture_log

  doctest MedianOfTwoSortedArrays

  test "example cases" do
    assert find_median_sorted_arrays([1, 3], [2]) == 2
    assert find_median_sorted_arrays([1, 2], [3, 4]) == 2.5
  end

  test "extra cases" do
    assert find_median_sorted_arrays([1], []) == 1
    assert find_median_sorted_arrays([], [1]) == 1
    assert find_median_sorted_arrays([1], [2]) == 1.5
    assert find_median_sorted_arrays([1, 3, 5], [2, 4]) == 3
    assert find_median_sorted_arrays([1, 3, 4, 5, 6], [2]) == 3.5
  end
end
